src/Features/Core/Portable/Completion/CompletionOptions.cs

```diff
@@ -2,201 +2,36 @@
-              HideAdvancedMembers: options.GetOption(Metadata.HideAdvancedMembers, language),
-              ShowNameSuggestions: options.GetOption(Metadata.ShowNameSuggestions, language),
-              ShowItemsFromUnimportedNamespaces: options.GetOption(Metadata.ShowItemsFromUnimportedNamespaces, language),
-              UnnamedSymbolCompletionDisabled: options.GetOption(Metadata.UnnamedSymbolCompletionDisabledFeatureFlag),
-              TargetTypedCompletionFilter: options.GetOption(Metadata.TargetTypedCompletionFilterFeatureFlag),
-              TypeImportCompletion: options.GetOption(Metadata.TypeImportCompletionFeatureFlag),
-              ProvideDateAndTimeCompletions: options.GetOption(Metadata.ProvideDateAndTimeCompletions, language),
-              ProvideRegexCompletions: options.GetOption(Metadata.ProvideRegexCompletions, language),
-              TimeoutInMillisecondsForExtensionMethodImportCompletion: options.GetOption(Metadata.TimeoutInMillisecondsForExtensionMethodImportCompletion));
-
-        public OptionSet WithChangedOptions(OptionSet set, string language)
-            => set.
-                WithChangedOption(Metadata.TriggerOnTyping, language, TriggerOnTyping).
-                WithChangedOption(Metadata.TriggerOnTypingLetters, language, TriggerOnTypingLetters).
-                WithChangedOption(Metadata.TriggerOnDeletion, language, TriggerOnDeletion).
-                WithChangedOption(Metadata.TriggerInArgumentLists, language, TriggerInArgumentLists).
-                WithChangedOption(Metadata.IsExpandedCompletion, IsExpandedCompletion).
-                WithChangedOption(Metadata.EnterKeyBehavior, language, EnterKeyBehavior).
-                WithChangedOption(Metadata.SnippetsBehavior, language, SnippetsBehavior).
-                WithChangedOption(Metadata.HideAdvancedMembers, language, HideAdvancedMembers).
-                WithChangedOption(Metadata.ShowNameSuggestions, language, ShowNameSuggestions).
-                WithChangedOption(Metadata.ShowItemsFromUnimportedNamespaces, language, ShowItemsFromUnimportedNamespaces).
-                WithChangedOption(Metadata.UnnamedSymbolCompletionDisabledFeatureFlag, UnnamedSymbolCompletionDisabled).
-                WithChangedOption(Metadata.TargetTypedCompletionFilterFeatureFlag, TargetTypedCompletionFilter).
-                WithChangedOption(Metadata.TypeImportCompletionFeatureFlag, TypeImportCompletion).
-                WithChangedOption(Metadata.ProvideDateAndTimeCompletions, language, ProvideDateAndTimeCompletions).
-                WithChangedOption(Metadata.ProvideRegexCompletions, language, ProvideRegexCompletions).
-                WithChangedOption(Metadata.TimeoutInMillisecondsForExtensionMethodImportCompletion, TimeoutInMillisecondsForExtensionMethodImportCompletion);
+        // note: must pass at least one parameter to avoid calling default ctor:
+        public static readonly CompletionOptions Default = new(TriggerOnTyping: true);
```

Comment:
ick :D

Reply:
Yeah, it's unfortunate. I think the best thing to do would be to invert all options that default to non-`default(T)` value, so that `default(CompletionOptions)` is the default.

Reply:
https://github.com/dotnet/roslyn/issues/59195